soc/arm/nordic_nrf/validate_rram_partitions.c

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/devicetree.h>
+#include <zephyr/sys/util_macro.h>
+#include <zephyr/toolchain.h>
+
+#define PAIR__(f, sep, arg_first, ...) FOR_EACH_FIXED_ARG(f, sep, arg_first, __VA_ARGS__)
+#define PAIR_(f, sep, args_to_expand)  PAIR__(f, sep, args_to_expand)
+#define PAIR(n, f, sep, ...)	       PAIR_(f, sep, GET_ARGS_LESS_N(n, __VA_ARGS__))
+
+/**
+ * @brief Call a macro on every unique pair of the given variadic arguments.
+ *
+ * For example, FOR_EACH_PAIR(f, (,), 1, 2, 3, 4) should expand to:
+ *
+ *     f(2, 1) , f(3, 1) , f(4, 1) , f(3, 2) , f(4, 2) , f(4, 3)
+ *
+ * @param f   Macro to call. Must accept two arguments.
+ * @param sep Separator between macro calls. Must be in parentheses.
+ *
+ * @see FOR_EACH
+ */
+#define FOR_EACH_PAIR(f, sep, ...)                                                                 \
+	LISTIFY(NUM_VA_ARGS_LESS_1(__VA_ARGS__), PAIR, sep, f, sep, __VA_ARGS__)
+
+/**
+ * @brief Get a node's non-secure register block start address.
+ *
+ * @param node_id Node identifier.
+ */
+#define REG_ADDR_NS(node_id) (DT_REG_ADDR(node_id) & 0xEFFFFFFFUL)
+
+/**
+ * @brief Get a node's non-secure register block end address.
+ *
+ * @param node_id Node identifier.
+ */
+#define REG_END_NS(node_id) (REG_ADDR_NS(node_id) + DT_REG_SIZE(node_id))
+
+/* clang-format off */
+
+#define RRAM_BASE REG_ADDR_NS(DT_NODELABEL(rram0))
+#define RRAM_CONTROLLER DT_NODELABEL(rram_controller)
+
+#if !DT_NODE_EXISTS(RRAM_CONTROLLER)
+#error "Missing \"rram-controller\" node"
```

Comment:
I think this would print as

`50 | #error "Missing \"rram-controller\" node"`

on the console. #error does not need quotes, just drop them all, keep it simple, the compiler quotes the whole line anyway

Reply:
@anangl please address in https://github.com/zephyrproject-rtos/zephyr/pull/68043

Reply:
Can also be addressed in #68309, which adds the same file again.